python/pysrc/kaskada/sources/arrow.py

```diff
@@ -109,35 +143,120 @@ def __init__(
         )
 
         self._convert_options = pyarrow.csv.ConvertOptions(column_types=schema)
-        self.add_rows(rows)
 
-    def add_rows(self, rows: dict | list[dict]) -> None:
+    @staticmethod
+    async def create(
+        rows: Optional[dict | list[dict]] = None,
+        *,
+        time_column: str,
+        key_column: str,
+        retained: bool = True,
```

Comment:
I didn't add the `retained` in all sources. That can be a follow up when we figure out the language